server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardIntegrationTest.java

```diff
@@ -333,6 +333,26 @@ void forwardShouldHandleLocalCopy() throws Exception {
         }));
     }
 
+    @Test
+    void forwardShouldKeepMessageWhenOnlyLocalCopy() throws Exception {
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(BOB),
+                Mapping.forward(BOB.asString()));
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(FROM, PASSWORD)
+            .sendMessage(FROM, BOB.asString());
+
+        Awaitility.await().until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1L);
+
+        SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+            Mail forwardedMail = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
+                .findAny().get();
+            softly.assertThat(forwardedMail.getRecipients()).containsOnly(BOB.asMailAddress());
+            softly.assertThat(forwardedMail.getMaybeSender().asOptional()).contains(BOB.asMailAddress());
```

Comment:
If we only have the local copy maybe we should not rewrite the sender?

Reply:
I agree, it should be fixed.